var/spack/repos/builtin/packages/py-nanotron/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyNanotron(PythonPackage):
+    """Minimalistic large language model 3D-parallelism training."""
+
+    homepage = "https://github.com/huggingface/nanotron"
+    url = "https://github.com/huggingface/nanotron/archive/refs/tags/v0.4.tar.gz"
+    git = "https://github.com/huggingface/nanotron.git"
+
+    maintainers("thomas-bouvier")
+
+    license("Apache-2.0")
+
+    version("main", branch="main")
+    version("0.4", sha256="7ea378eb1b6b16c93a3021fcfd71dd73bd14d826126eae215e2b02e05cd6a120")
```

Comment:
```
$ spack checksum --verify --batch py-nanotron 0.4
==> Found 1 version of py-nanotron
==> Fetching https://github.com/huggingface/nanotron/archive/refs/tags/v0.4.tar.gz
==> Verified 0 of 1

  0.4  [x] 30e9cdd07e86166dd9690351d9d995b3560810044fdca64737ed042cd91c792a


==> Error: Invalid checksums found.
```
Fixed by
```suggestion
    version("0.4", sha256="30e9cdd07e86166dd9690351d9d995b3560810044fdca64737ed042cd91c792a")
```

Reply:
Also, 0.5 available as tag, but not listed as release on GitHub, if you want to add it.
```
    version("0.5", sha256="73f7127c6f7af6a5f031843575f0621b392f8a7c91462b78d1460c8a586f04f0")
```

Reply:
Oops, I had originally used the hash from the PyPi archive. Fixed.

Reply:
[I inquired](https://github.com/huggingface/nanotron/issues/268) about the availability of a release 0.5 as it isn't listed on PyPi either. We can proceed with integrating this PR, and I'll create a new one based on the response.